metrics/web/test/base/presentation/widgets/svg_image_test.dart

```diff
@@ -0,0 +1,351 @@
+          return tester.pumpWidget(
+            _SvgImageTestbed(
+              defaultSrc,
+              fit: null,
+              rendererHelper: rendererHelperMock,
+            ),
+          );
+        });
+
+        final image = tester.widget<Image>(find.byType(Image));
+
+        expect(image.fit, equals(BoxFit.none));
+      },
+    );
+
+    testWidgets(
+      "applies the given height to the network image widget when not using the SKIA renderer",
+      (WidgetTester tester) async {
+        const height = 20.0;
+        when(rendererHelperMock.isSkia).thenReturn(false);
+
+        await mockNetworkImagesFor(() {
+          return tester.pumpWidget(
+            _SvgImageTestbed(
+              defaultSrc,
+              height: height,
+              rendererHelper: rendererHelperMock,
+            ),
+          );
+        });
+
+        final image = tester.widget<Image>(find.byType(Image));
+
+        expect(image.height, equals(height));
+      },
+    );
+
+    testWidgets(
+      "applies the given width to the network image widget when not using the SKIA renderer",
+      (WidgetTester tester) async {
+        const width = 20.0;
+        when(rendererHelperMock.isSkia).thenReturn(false);
+
+        await mockNetworkImagesFor(() {
+          return tester.pumpWidget(
+            _SvgImageTestbed(
+              defaultSrc,
+              width: width,
+              rendererHelper: rendererHelperMock,
+            ),
+          );
+        });
+
+        final image = tester.widget<Image>(find.byType(Image));
+
+        expect(image.width, equals(width));
+      },
+    );
+
+    testWidgets(
+      "applies the given color to the network image widget when not using the SKIA renderer",
+      (WidgetTester tester) async {
+        const color = Colors.red;
+        when(rendererHelperMock.isSkia).thenReturn(false);
+
+        await mockNetworkImagesFor(() {
+          return tester.pumpWidget(
+            _SvgImageTestbed(
+              defaultSrc,
+              rendererHelper: rendererHelperMock,
+              color: color,
+            ),
+          );
+        });
+
+        final image = tester.widget<Image>(find.byType(Image));
+
+        expect(image.color, equals(color));
+      },
+    );
+  });
+}
+
+/// A testbed class needed to test the [SvgImage] widget.
+class _SvgImageTestbed extends StatelessWidget {
+  /// A path to an image to use in tests.
+  final String src;
+
+  /// A height of an image to use in tests.
+  final double height;
+
+  /// A width of an image to use in tests.
+  final double width;
+
+  /// A parameter that controls how to inscribe an image into the space
+  /// allocated during layout.
+  final BoxFit fit;
+
+  /// A [Color] used to combine with an image to use in tests.
+  final Color color;
+
+  /// A [RendererHelper] to use in tests.
+  final RendererHelper rendererHelper;
+
+  /// Creates a new instance of the svg image testbed.
+  const _SvgImageTestbed(
+    this.src, {
```

Comment:
Would it be great to have a default `src` here to be able to omit `src` when we do not need it in the test?